src/devices/BrickPi3/BrickPi3.cs

```diff
@@ -0,0 +1,948 @@
+    /// </summary>
+    public class Brick
+    {
+        private const int ChipSelectLine = 1;
+        // To store the Sensor types as well as date to be sent when it's an I2C sensor
+        private SensorType[] _sensorType = { SensorType.None, SensorType.None, SensorType.None, SensorType.None };
+        private byte[] _i2cInBytes = { 0, 0, 0, 0 };
+        //Initernals to initalize the SPI
+        private static SpiDevice _brickPiSPI = null;
+        private BrickPiVoltage _brickPiVoltage = new BrickPiVoltage();
+
+        #region Properties
+
+        /// <summary>
+        /// used to store the SPI Address
+        /// used mainly when multiple bricks in a raw or not the default SPI address 
+        /// up to 254 addresses supported
+        /// </summary>
+        public byte SPIAddress { get; set; }
+
+        /// <summary>
+        /// Stores all the information regarding the Brick
+        /// This includes id, hardware and firmware versions
+        /// </summary>
+        public BrickPiInfo BrickPi3Info { get; internal set; }
+
+        /// <summary>
+        /// Stores the voltage information
+        /// </summary>
+        public BrickPiVoltage BrickPi3Voltage
+        {
+            get
+            {
+                _brickPiVoltage.Voltage3V3 = GetVoltage3V3();
+                _brickPiVoltage.Voltage5V = GetVoltage5V();
+                _brickPiVoltage.Voltage9V = GetVoltage9V();
+                _brickPiVoltage.VoltageBattery = GetVoltageBatteryVcc();
+                return _brickPiVoltage;
+            }
+        }
+
+        #endregion
+
+        #region Init and reset
+
+        /// <summary>
+        /// Initialize the brick including SPI communication
+        /// </summary>
+        /// <param name="spiAddress"></param>
+        public void InitSPI(byte spiAddress = 1)
+        {
+            try
+            {
+                SPIAddress = spiAddress;
+                // SPI 0 is used on Raspberry with ChipSelectLine 1
+                var settings = new SpiConnectionSettings(0, ChipSelectLine);
+                // 500K is the SPI communication with the brick
+                settings.ClockFrequency = 500000;
+                // see http://tightdev.net/SpiDev_Doc.pdf
+                settings.Mode = SpiMode.Mode0;
+                settings.DataBitLength = 8;
+                // as the SPI is a static, checking if it has already be initialised
+                if (_brickPiSPI == null)
+                {
+                    _brickPiSPI = new UnixSpiDevice(settings);
+                }
+                BrickPi3Info = new BrickPiInfo();
+                BrickPi3Info.Manufacturer = GetManufacturer();
+                BrickPi3Info.Board = GetBoard();
+                BrickPi3Info.HardwareVersion = GetHardwareVersion();
+                BrickPi3Info.SoftwareVersion = GetFirmwareVersion();
+                BrickPi3Info.Id = GetId();
+            }
+            catch (Exception ex)
```

Comment:
any particular reason for eating the exception? I'd just let it flow unless this is intentional and it is possible to safely continue